### ISSUE TITLE
[vault-monitoring] Parametrize Alert Rule Cluster Size

### ISSUE THREAD

**Issue by in0rdr** [adfinis-sygroup/helm-charts#611]
This issue can be used to implement a parameter for specifying the cluster size and configure the alert thresholds in `vault-monitoring`.

**Describe the solution you'd like**

The alert thresholds (currently mostly `5`) in the Prometheus/Alertmanager ruleset should be configurable for other cluster sizes w/ node count different from 5.

For [example](https://github.com/adfinis-sygroup/helm-charts/blob/master/charts/vault-monitoring/values.yaml#L52):
```
  rules:
    - alert: VaultLeadershipLoss
      expr: sum(increase(vault_core_leadership_lost_count{job="vault-monitoring"}[1h])) > 5
```

This number `5` should be made configurable.

**Describe alternatives you've considered**

none so far

**Affected chart**

vault-monitoring

**Comment by in0rdr** [comment]
@pree not sure if that's even possible, I've never parametrized values in values.yaml.

This would need some kind of "meta" language or templating, not?

Because values.yaml is the original input (i.e., not templated)..

**Comment by hairmare** [comment]
> This would need some kind of "meta" language or templating, not?

Helm has a `tpl` operator for "meta" templating, that can be used to render templte strings from values.

In this case we should probably look into rewritig the rule in a more generic fashion first. Most likely there is a way to derive `5` from some other vector so we don't need to template it.

**Comment by eyenx** [comment]
the problem is already fixed. see #610

**Comment by in0rdr** [comment]
Thanks for the fast feedback @hairmare. @eyenx took care of the rewriting. I really like this suggestions and I will test it thoroughly in #610, but it will most probably "just work" (without explicit threshold).

I noticed that I pasted the wrong example code in the beginning of this issue (the `5` in this example probably does not refer to the client count). Anyways, thanks for the help. Consider this closed.

But certainly, if there are obvious ways to replace the other threshold in a smart way we should probably do this.

Thanks
